code/game/gamemodes/miniantags/demons/slaughter demon/newSlaughter.dm

```diff
@@ -0,0 +1,284 @@
+	var/channeling = FALSE
+	var/lasthealtime = 0
+	var/phased = TRUE
+	var/phaseoutchaneltime = 0
+	var/phaseinchaneltime = 0
+	var/stat_amplification = 0.5
+	var/max_amplification_stacks = 0.5 //50% max
+	var/last_stat_decrease = 0
+	var/decay_time = 0
+	var/boost = 0
+	var/feast_sound = 'sound/misc/demon_consume.ogg'
+	var/devoured = 0
+	var/mindsabsorbed = 0
+	var/list/consumed_mobs = list()
+	var/list/nearby_mortals = list()
+	var/list/bloodspots = list()
+	var/cooldown = 0
+	var/gorecooldown = 0
+	var/vialspawned = FALSE
+	var/turf/channel_target = 0
+	var/obj/effect/proc_holder/spell/demon_slam/slam_holder
+	var/obj/effect/proc_holder/spell/demon_charge/charge_holder
+	loot = list(/obj/effect/decal/cleanable/blood/innards, /obj/effect/decal/cleanable/blood, /obj/effect/gibspawner/generic, /obj/effect/gibspawner/generic, /obj/item/organ/internal/heart/demon/slaughter)
+	var/playstyle_string = "<B>You are the Slaughter Demon, a terrible creature from another existence. You have a single desire: to kill.  \
+						You may use the blood crawl icon when on blood pools to travel through them, appearing and dissapearing from the station at will. \
+						Pulling a dead or critical mob while you enter a pool will pull them in with you, allowing you to feast. \
+						You move quickly upon leaving a pool of blood, but the material world will soon sap your strength and leave you sluggish. </B>"
+	del_on_death = TRUE
+	deathmessage = "screams in anger as it collapses into a puddle of viscera!"
+
+/mob/living/simple_animal/demon/slaughter_demon/New()
+	..()
+	remove_from_all_data_huds()
+	var/obj/effect/proc_holder/spell/bloodcrawldemon/bloodspell = new
+	var/obj/effect/proc_holder/spell/demon_charge/dc = new
+	var/obj/effect/proc_holder/spell/demon_slam/ds = new
+	charge_holder = dc
+	slam_holder = ds
+	AddSpell(dc)
+	AddSpell(ds)
+	AddSpell(bloodspell)
+	if(istype(loc, /obj/effect/dummy/slaughter_demon))
+		phased = FALSE
+	addtimer(CALLBACK(src, PROC_REF(attempt_objectives)), 5 SECONDS)
+
+/obj/effect/dummy/slaughter_demon //Can't use the wizard one, blocked by jaunt/slow
+	name = "odd blood"
+	icon = 'icons/effects/effects.dmi'
+	icon_state = "nothing"
+	density = FALSE
+	anchored = TRUE
+	invisibility = 60
+	resistance_flags = LAVA_PROOF | FIRE_PROOF | UNACIDABLE | ACID_PROOF
+
+/obj/effect/dummy/slaughter_demon/relaymove(mob/user, direction)
+	forceMove(get_step(src, direction))
+
+/obj/effect/dummy/slaughter_demon/ex_act()
+	return
+
+/obj/effect/dummy/slaughter_demon/bullet_act()
+	return
+
+/obj/effect/dummy/slaughter_demon/singularity_act()
+	return
+
+/*/obj/item/antag_spawner/slaughter_demon/spawn_antag(client/C, turf/T, type = "", mob/user)
+	var/obj/effect/dummy/slaughter/holder = new /obj/effect/dummy/slaughter(T)
+	var/mob/living/simple_animal/demon/D = new demon_type(holder)
+	if(istype(D, /mob/living/simple_animal/demon/slaughter))
+		var/mob/living/simple_animal/demon/slaughter/S = D
+		S.vialspawned = TRUE*/
```

Comment:
whats this about?

Reply:
Reference code for me to look at, will remove it with the next commit, definitely shouldn't be here